### PR TITLE
If no socket is attached to error port, log to console.error

### PR DESCRIPTION
**Pull request by catch-point**

### CODE REVIEW

File: test/promise-component-mocha.js [hunk 1]
```diff
@@ -118,6 +118,32 @@ describe('promise-component', function() {
             return onceData(component, 'outgoing', 'sentback', 'incoming', "hello");
         }).should.be.rejectedWith("hello world");
     });
+    it("should log rejections to console.error when no reject port attached", function() {
```
Comment (glenna-m) [inline test/promise-component-mocha.js:121]: I do not follow how you are forcing the reject port to not be attached.  That port is created inside promise-component, no?

File: test/promise-component-mocha.js [hunk 2]
```diff
@@ -118,6 +118,32 @@ describe('promise-component', function() {
...
+                'in':{
+                    ondata: function(payload) {
+                        throw payload + " world";
+                    }
+                }
+            }
+        }).then(promiseComponent).then(createComponent).then(function(component){
+            var console_error = console.error;
+            return new Promise(function(resolve, reject) {
+                console.error = reject;
```
Comment (glenna-m) [inline test/promise-component-mocha.js:133]: Also do not follow how this really tests console logging.  Perhaps we should go over this on Monday.

File: test/promise-component-mocha.js (hunk 1 above)
Comment (catch-point) [inline test/promise-component-mocha.js:121]: The outport is created, but in this test nothing is attached to it (only the in/out ports are attached).

**Comment by glenna-m** [comment]
After discussion with James, this looks good.  LGTM to merge.

**Pull request merged**